Add Namespace support to the API module and the CLI commands (#6874)
hashicorp/consul: website/source/api/acl/auth-methods.html.md

  the chosen auth method. Contents will vary depending upon the type chosen.
  For more information on configuring specific auth method types, see the [auth
  method documentation](/docs/acl/acl-auth-methods.html).
  
- `Namespace` `(string: "")` - **Enterprise Only** Specifies the namespace to 
  create the Auth Method within. If not provided in the JSON body, the value of
  the `ns` URL query parameter or the `X-Consul-Namespace` header will be used. 

>   If not provided at all, the namespace will be inferred from the request's ACL 

Reviewer: ```suggestion
  If not provided at all, the namespace will be inherited from the request's ACL 
```

>   token, or will default to the `default` namespace. Added in Consul 1.7.0.

Reviewer: ```suggestion
  token or will be set to the `default` namespace. Added in Consul 1.7.0.
```

- `name` `(string: <required>)` - Specifies the name of the ACL auth method to
  read. This is required and is specified as part of the URL path.
  

> - `ns` `(string: "")` - **Enterprise Only** Specified the namespace to lookup

Reviewer: ```suggestion
- `ns` `(string: "")` - **Enterprise Only** Specifies the namespace to lookup within
```

>   the Auth Method within. This value can be specified as the `ns` URL query 

Reviewer: ```suggestion
  the Auth Method. This value can be specified as the `ns` URL query 
```
Should Auth Method be capitalized?

>   parameter, or the `X-Consul-Namespace` header. If not provided by either,

Reviewer: ```suggestion
  parameter or in the `X-Consul-Namespace` header. If not provided by either,
```

>   the namespace will be inferred from the request's ACL token, or will default

Reviewer: ```suggestion
  the namespace will be inherited from the request's ACL token or will be set
```

  the chosen auth method. Contents will vary depending upon the type chosen.
  For more information on configuring specific auth method types, see the [auth
  method documentation](/docs/acl/acl-auth-methods.html).
  
- `Namespace` `(string: "")` - **Enterprise Only** Specifies the namespace of
  the Auth Method to update. If not provided in the JSON body, the value of
  the `ns` URL query parameter or the `X-Consul-Namespace` header will be used. 

>   If not provided at all, the namespace will be inferred from the request's ACL 

Reviewer: If you accept the previous changes, they should be applied here too.

- `name` `(string: <required>)` - Specifies the name of the ACL auth method to
  delete. This is required and is specified as part of the URL path.
  

> - `ns` `(string: "")` - **Enterprise Only** Specified the namespace of the

Reviewer: Same as above, changes should be propagated.
